LiskHQ/lisk-hub
# Use new subscribeToDeviceConnceted and subscribeToDeviceDisconnceted utils - closes #2442

… the hwManager communications file



### What issue have I solved?

-- #2442 

### How have I implemented/fixed it?

This PR is for refactor the existing code for detect when a hardware wallet device is connect or disconnect, so the following files were updated.

- [x] libs/hwManager/communication.js
- [x] src/store/middlewares/hwWallet.js

#### 1. Communication.js file
This file was update to use the correct names for the IPC channel names (constants), the previous ones were, CONNECT and DISCONNECT and replace by HW_CONNECTED and HW_DISCONNECTED, after this change, the react side was able to receive properly the data coming from these functions.

#### 2.hwWallet.js
This file was rename from hwWallet.js to hwManager.js to keep consistency with the new code and because of this the middleware index.js file was updated with the new name.

In the other hand 2 functions where imported subscribeToDeviceConnceted and subscribeToDeviceDisonnceted from utils/hwManager.js file that is the one that has all hw manager functions to interact.

In the same file I remove the unused code and add jsDoc to the functions for better understanding.

### How has this been tested?

The way for test that the functionality is working as should, the following steps needs to be follow/

1. Run the app in electron.
2. connect any hw device.
3. A green toaster message should show up with the model of the connected device.
4. When disconnect a device a green toaster message should show up with the device model
5. Connect again any hw device and login to the app.
6. You can send or vote, all this kind of transactions should be working as nothing change.
7. Remove the hw device that you used for login.
8. A dialog should show up and then after click in the button log you out.
9. A green toaster will show up with the device model (as soon you remove the device).

NOTE: All this manual test is only for check that connect/disconnect functionality of the new code.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)
